eng/pipelines/update-azure-sdk-for-net-codes.yml

```diff
@@ -0,0 +1,53 @@
+# Generate SDK codes
+
+parameters:
+  name: ''
+  filter: ''
+
+jobs:
+- job: ${{ parameters.name }}
+  timeoutInMinutes: 60
+  pool:
+    vmImage: ubuntu-20.04
+  variables:
+    AutorestCSharpVersion: $[stageDependencies.Build_and_Test.Build.outputs['Package.AutorestCSharpVersion']]
+    CadlEmitterVersion: $[stageDependencies.Build_and_Test.Build.outputs['Publish.CadlEmitterVersion']]
+  dependsOn: Update_Generator_Versions
+  steps:
+  - checkout: self
+    fetchDepth: 1
+  - checkout: azure-sdk-for-net
+  - checkout: azure-sdk-tools
+    fetchDepth: 1
+  - task: UseDotNet@2
+    displayName: 'Use .NET Core SDK'
+    inputs:
+      useGlobalJson: true
+      performMultiLevelLookup: true
+  - task: Powershell@2
+    displayName: Update SDK codes ${{ parameters.filter }}
+    inputs:
+      pwsh: true
+      filePath: $(Build.SourcesDirectory)/autorest.csharp/eng/UpdateAzureSdkForNet.ps1
+      arguments: >
+        -AutorestCSharpVersion $(AutorestCSharpVersion)
+        -CadlEmitterVersion $(CadlEmitterVersion)
+        -SdkRepoRoot $(Build.SourcesDirectory)/azure-sdk-for-net
+        -ServiceDirectoryFilters ${{ parameters.filter}}
+        -ShowSummary
+      failOnStderr: false
+      workingDirectory: $(Build.SourcesDirectory)/autorest.csharp
+    retryCountOnTaskFailure: 3
+#   - pwsh: |
+#       git restore eng/Packages.Data.props
+#       git restore eng/emitter-package.json
+#     workingDirectory: $(Build.SourcesDirectory)/azure-sdk-for-net
+  - template: /eng/common/pipelines/templates/steps/git-push-changes.yml@azure-sdk-tools
```

Comment:
Why are each of these templates pushing changes? It looks like they are pushing changes to the same branch even. I wouldn't expect we would push any changes unless all the validation has completed.

Reply:
We are doing this partially to support PRs with 5000+ files changed.  By splitting these into commits we can still review each commit since as we know reviewing a PR with 5000+ files doesn't work in pretty much any tool out there.

Reply:
I believe I'm starting to understand why you are pushing for each job but I will add that this potentially runs the risk of conflicts if files under a shared path change.

Reply:
Agreed that risk is there, but that would break our conventions so if it fails in that case I think I am ok with that.  Worst case we can hard code common paths to run first.

Reply:
> I believe I'm starting to understand why you are pushing for each job but I will add that this potentially runs the risk of conflicts if files under a shared path change.

> Agreed that risk is there, but that would break our conventions so if it fails in that case I think I am ok with that. Worst case we can hard code common paths to run first.

AFAIKS, there are 2 ways to resolve that:
1. If it's a rare case, we can ask developers to manually create a PR to update `azure-sdk-for-net`, which we have done before.
2. Or we change to use our own scripts to push changes to github. The overlapped part should have the same changes. I did a test on local, such conflict can be resolved by `git pull --rebase`.